src/Features/Core/Portable/ExternalAccess/UnitTesting/API/NewUnitTestingIncrementalAnalyzer.cs

```diff
@@ -0,0 +1,88 @@
+    {
+        private readonly INewUnitTestingIncrementalAnalyzerImplementation _implementation;
+
+        public NewUnitTestingIncrementalAnalyzer(INewUnitTestingIncrementalAnalyzerImplementation implementation)
+            => _implementation = implementation;
+
+        public Task AnalyzeDocumentAsync(Document document, SyntaxNode bodyOpt, UnitTestingInvocationReasons reasons, CancellationToken cancellationToken)
+            => _implementation.AnalyzeDocumentAsync(document, bodyOpt, reasons, cancellationToken);
+
+        public Task AnalyzeProjectAsync(Project project, bool semanticsChanged, UnitTestingInvocationReasons reasons, CancellationToken cancellationToken)
+            => _implementation.AnalyzeProjectAsync(project, semanticsChanged, reasons, cancellationToken);
+
+        public Task AnalyzeSyntaxAsync(Document document, UnitTestingInvocationReasons reasons, CancellationToken cancellationToken)
+            => _implementation.AnalyzeSyntaxAsync(document, reasons, cancellationToken);
+
+        public Task DocumentCloseAsync(Document document, CancellationToken cancellationToken)
+            => Task.CompletedTask;
+
+        public Task DocumentOpenAsync(Document document, CancellationToken cancellationToken)
+            => Task.CompletedTask;
+
+        public Task DocumentResetAsync(Document document, CancellationToken cancellationToken)
+            => Task.CompletedTask;
+
+        public Task ActiveDocumentSwitchedAsync(TextDocument document, CancellationToken cancellationToken)
+            => Task.CompletedTask;
+
+        public Task NewSolutionSnapshotAsync(Solution solution, CancellationToken cancellationToken)
+            => Task.CompletedTask;
+
+        public Task RemoveDocumentAsync(DocumentId documentId, CancellationToken cancellationToken)
+        {
+            _implementation.RemoveDocument(documentId);
+            return Task.CompletedTask;
+        }
+
+        public Task RemoveProjectAsync(ProjectId projectId, CancellationToken cancellationToken)
+            => Task.CompletedTask;
+
+        public Task NonSourceDocumentOpenAsync(TextDocument textDocument, CancellationToken cancellationToken)
+            => Task.CompletedTask;
+
+        public Task NonSourceDocumentCloseAsync(TextDocument textDocument, CancellationToken cancellationToken)
+            => Task.CompletedTask;
+
+        public Task NonSourceDocumentResetAsync(TextDocument textDocument, CancellationToken cancellationToken)
+            => Task.CompletedTask;
+
+        public Task AnalyzeNonSourceDocumentAsync(TextDocument textDocument, UnitTestingInvocationReasons reasons, CancellationToken cancellationToken)
+            => Task.CompletedTask;
+
+        public void LogAnalyzerCountSummary()
+        {
+        }
+
+        /// <summary>
+        /// Order all incremental analyzers below DiagnosticIncrementalAnalyzer
+        /// </summary>
+        public int Priority => 1;
```

Comment:
Is a concept of priority still relevant given that UT's analyzer will be the only one going forward?

Reply:
no, it is not.  my followup PR removes a TON of this, but commit by commit so ti can be reasoned about.